### ISSUE TITLE
Documentation for config files

### ISSUE THREAD

**Issue by rpatters1** [Nick-Mazuk/jw-lua-scripts#73]
We're making good progress on documenting the library functions. I wonder is there is an equivalent way to capture the config file options. There should probably be a documentation page for how to create and use them, at the very least.

**Comment by Nick-Mazuk** [comment]
This sounds like a great idea!

I'm not entirely sure how to capture the config options unless we standardize the syntax for it. When I work on creating a downloadable version of each script (https://github.com/Nick-Mazuk/jw-lua-scripts/issues/70), I'll try to figure this out. I'm hoping to be able to be able to list every script in the repo on the site, so people can just download the scripts straight from the site without ever touching GitHub. I'll add the config options there.

In a similar vain, would it make sense to create a separate script that allows users to update the configuration files for any of the scripts? That way, they won't need to manually go to the txt file to update the values.

In the meantime, I fully support adding a documentation page for creating and using the configuration files. You just need to add a markdown file to the `/docs` folder. For instance, `/docs/configuration.md`.

**Comment by Nick-Mazuk** [comment]
Does this page resolve this issue?

https://www.finalelua.com/docs/rgp-lua/rgp-lua-configuration

**Comment by rpatters1** [comment]
No. That's for RGP Lua. I was thinking about the optional config files in lib/configuration.lua

**Comment by Nick-Mazuk** [comment]
Ok. Here's one shower thought I had if RGP Lua could set environment variables.

For instance, let's say we had this pseudocode:

```lua
local file = "my_script.config.txt"
if process.env.REGISTER_CONFIGURATION then
    for key, value in pairs(config) do
        -- `rgplua` is a namespace specifically for
        -- RGP Lua similar to `finenv` or `finale`.
        rgplua.RegisterConfigParameter(file, key, value)
    end
end

-- Later in the code...
-- If sandbox mode is on, this is not strictly necessary
-- but could be useful if the script is expensive to run
if process.env.RUN_SCRIPT then
    do_main()
end
```

Then if RGP Lua had a way of setting the configs for a particular script in the plugin itself, it could silently run the script behind the scenes and see which config values were registered. Preferably it would also set sandbox mode to `true`.

Since all configs must eventually call `configuration.get_get_parameters(file_name, parameter_list)`, we could add this registration to that function behind the scenes.

**Comment by rpatters1** [comment]
I'm having trouble understanding exactly the issue you want to address, but in general I think we should keep RGP Lua simple and do things in Lua code to the extent possible.

Could this be an enhancement of the configuration library itself? Or a separate library?

For what it's worth, "sandbox mode" was never actually implemented in the PDK Framework as far as I can tell. Instead, `NoStore` simply forces a blanket rollback of any changes that were made by the script. Lua scripts can do this as easily as the PDK Framework can.
